### PR TITLE
Adapt `spring-boot-example-app` to Kubernetes

### PR DESCRIPTION
**Pull request by Sgitario**
The major change is that by default the `spring-boot-example-app` chart will NOT expose the application either in OCP or K8s. 
So, users using the new version 0.0.4 would need to add `route.expose=true` if they want to expose it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,16 +49,26 @@ snowdrop	    https://snowdrop.github.io/helm
...
 Requirements:
-- Connected/logged to a Openshift cluster
+- Connected/logged to a Kubernetes/Openshift cluster
 - Have installed [the Helm command line](https://helm.sh/docs/intro/install/)
 
 ### spring-boot-example-app
 
-This chart deploys and exposes a Spring Boot application in Openshift.
-To install it, you need to execute the following command:
+This chart deploys and exposes a Spring Boot application in Kubernetes or Openshift.
```
Comment (cmoulliard) [inline README.md]: `in Kubernetes or Openshift.` --> `on Kubernetes or OpenShift.`

Remark: Openshift name is `OpenShift` and not `Openshift` ;-)

File: README.md [hunk 2]
```diff
@@ -49,16 +49,26 @@ snowdrop	    https://snowdrop.github.io/helm
...
+
+- For Kubernetes:
+To use it in Kubernetes, the image of the Spring Boot application needs to be available in a public container registry. In this example, we'll use the image `quay.io/user/my-app:latest`. To install it, you need to execute the following command:
```
Comment (cmoulliard) [inline README.md]: `...it in kubrnetes...` --> `on kubernetes` 
To be changed everywhere !

and

`to be available in a public image registry` --> `to be published on an images registry where you have access and be logged`

File: README.md [hunk 3]
```diff
@@ -49,16 +49,26 @@ snowdrop	    https://snowdrop.github.io/helm
...
 
 ```console
-$ helm install my-spring-boot-app snowdrop/spring-boot-example-app --set name=app --set s2i.source.repo=http://github.com/org/repo --set s2i.source.ref=main
+$ helm install my-spring-boot-app snowdrop/spring-boot-example-app --set name=app --set docker.image=quay.io/user/my-app:latest --set ingress.host=<your kubernetes domain>
+```
+
+**note**: if you want to expose your application in Kubernetes, you need to provide the `ingress.host` property.
+
+- For Openshift, to install it, you need to execute the following command:
```
Comment (cmoulliard) [inline README.md]: Take care about the OpenShift name please

File: README.md [hunk 4]
```diff
@@ -146,18 +156,32 @@ $ helm dependency update
 
 After doing this, helm will download the `spring-boot-example-app` chart from the repository `http://snowdrop.github.io/helm` and will copy the Chart tallball `spring-boot-example-app-0.0.3.tgz` at `my-custom-chart/charts/`.
```
Comment (cmoulliard) [inline README.md:157]: `After doing this, helm will download the `spring-boot-example-app` chart from the repository ...`. Are you sure ? This is not correct from my point of view to say that as first `helm install` is gonna to deploy using the chart the K8S manifests on the cluster. When you use OpenShift, it is needed to deploy a BuildConfig and Build resources which are responsible on OpenShift to trigger the creation of a pod doing the S2I build. To be reviewed

File: README.md [hunk 5]
```diff
@@ -146,18 +156,32 @@ $ helm dependency update
...
 
-As we explained in [the usage section](#spring-boot-example-app) of the `spring-boot-example-app` chart, we need to provide the GIT repository and branch, so Openshift can build and deploy the Spring Boot application using S2i. For doing this, we need to edit the `my-custom-chart/values.yaml` file and provide this configuration:
+As we explained in [the usage section](#spring-boot-example-app) of the `spring-boot-example-app` chart, for Openshift we need to provide the GIT repository and branch, so Openshift can build and deploy the Spring Boot application using S2i. For doing this, we need to edit the `my-custom-chart/values.yaml` file and provide this configuration:
```
Comment (cmoulliard) [inline README.md]: Text will be more readable if you say something like this ==> "As the build is taking place on the cluster using OpenShift - see usage section, then we have to configure the `s2i` fields defined within the `values file ..."

File: README.md [hunk 6]
```diff
@@ -146,18 +156,32 @@ $ helm dependency update
...
   version: 0.0.1
+  route:
+    expose: true
   s2i:
     source:
       repo: http://github.com/org/repo
       ref: main
 ```
 
+If you are using Kubernetes, then you would only need to provide the docker image in the `my-custom-chart/values.yaml` file:
```
Comment (cmoulliard) [inline README.md]: How will the image be build in this case ?

File: README.md [hunk 7]
```diff
@@ -238,7 +266,7 @@ helm install crud snowdrop/spring-boot-example-crud
 
 ### spring-boot-example-configmap
 
-This chart deploys the example from [the repository](https://github.com/snowdrop/configmap-example). This example uses a ConfigMap to externalize configuration. ConfigMap is an object used by OpenShift to inject configuration data as simple key and value pairs into one or more Linux containers while keeping the containers independent of OpenShift.
+This chart deploys the example from [the repository](https://github.com/snowdrop/configmap-example). This example uses a ConfigMap to externalize configuration. ConfigMap is an object used by OpenShift and Kubernetes to inject configuration data as simple key and value pairs into one or more Linux containers while keeping the containers independent.
```
Comment (cmoulliard) [inline README.md]: "... to externalize the configuration". Idem for "The ConfigMap...."

and

"while keeping the containers independent." From what ?

File: README.md (hunk 1 above)
Comment (Sgitario) [reply README.md]: Changed

File: README.md (hunk 2 above)
Comment (Sgitario) [reply README.md]: changed

File: README.md (hunk 3 above)
Comment (Sgitario) [reply README.md]: changed

File: README.md (hunk 4 above)
Comment (Sgitario) [reply README.md:157]: This sentence does not mention OpenShift and explains what the command `helm dependency update` does. 
What happens with OpenShift or Kubernetes, it's described in a previous section.

File: README.md (hunk 5 above)
Comment (Sgitario) [reply README.md]: changed

File: README.md (hunk 6 above)
Comment (Sgitario) [reply README.md]: Rephrased

File: README.md (hunk 7 above)
Comment (Sgitario) [reply README.md]: I deleted this sentence as I don't think it adds value here.

**Pull request merged**